### PR TITLE
Vs2 functionality and build

### PR DESCRIPTION
**Pull request by baransrc**
This PR covers all the crucial parts of VS2, along with functional god mode, combat, scene structure, and win/lose condition.

### CODE REVIEW

File: Source/src/core/ErrorHandler.h [hunk 1]
```diff
@@ -90,8 +91,9 @@ namespace Hachiko
             }
             else
             {
-                HE_LOG(msg.c_str());
+                // HE_LOG(msg.c_str());
```
Comment (baransrc) [inline Source/src/core/ErrorHandler.h:94]:
```suggestion
                HE_LOG(msg.c_str());
```

File: Source/src/core/GameObject.cpp [hunk 2]
```diff
@@ -271,7 +271,7 @@ void Hachiko::GameObject::Update()
 
 void Hachiko::GameObject::DrawAll(ComponentCamera* camera, Program* program) const
 {
-    OPTICK_CATEGORY("Draw", Optick::Category::Rendering);
+    //OPTICK_CATEGORY("Draw", Optick::Category::Rendering);
```
Comment (baransrc) [inline Source/src/core/GameObject.cpp:274]:
```suggestion
    OPTICK_CATEGORY("Draw", Optick::Category::Rendering);
```

File: Source/src/core/GameObject.cpp [hunk 3]
```diff
@@ -284,7 +284,7 @@ void Hachiko::GameObject::DrawAll(ComponentCamera* camera, Program* program) con
 
 void Hachiko::GameObject::Draw(ComponentCamera* camera, Program* program) const
 {
-    OPTICK_CATEGORY("Draw", Optick::Category::Rendering);
+    //OPTICK_CATEGORY("Draw", Optick::Category::Rendering);
```
Comment (baransrc) [inline Source/src/core/GameObject.cpp:287]:
```suggestion
    OPTICK_CATEGORY("Draw", Optick::Category::Rendering);
```

File: Source/src/core/Scene.cpp [hunk 4]
```diff
@@ -324,6 +324,6 @@ void Hachiko::Scene::Start() const
 
 void Hachiko::Scene::Update() const
 {
-    OPTICK_CATEGORY("UpdateScene", Optick::Category::Scene);
+    //OPTICK_CATEGORY("UpdateScene", Optick::Category::Scene);
```
Comment (baransrc) [inline Source/src/core/Scene.cpp:327]:
```suggestion
    OPTICK_CATEGORY("UpdateScene", Optick::Category::Scene);
```

File: Source/src/core/rendering/Skybox.cpp [hunk 5]
```diff
@@ -40,7 +40,7 @@ Hachiko::Skybox::~Skybox()
 void Hachiko::Skybox::Draw(ComponentCamera* camera) const
 {
     // Use for optimized version (draw at the end) glDepthFunc(GL_LEQUAL);
-    OPTICK_CATEGORY("Draw", Optick::Category::Rendering);
+    //OPTICK_CATEGORY("Draw", Optick::Category::Rendering);
```
Comment (baransrc) [inline Source/src/core/rendering/Skybox.cpp:43]:
```suggestion
    OPTICK_CATEGORY("Draw", Optick::Category::Rendering);
```

File: Source/src/modules/ModuleRender.cpp [hunk 6]
```diff
@@ -193,7 +193,7 @@ UpdateStatus Hachiko::ModuleRender::Update(const float delta)
 
 void Hachiko::ModuleRender::Draw(Scene* scene, ComponentCamera* camera, ComponentCamera* culling)
 {
-    OPTICK_CATEGORY("Draw", Optick::Category::Rendering);
+    //OPTICK_CATEGORY("Draw", Optick::Category::Rendering);
```
Comment (baransrc) [inline Source/src/modules/ModuleRender.cpp:196]:
```suggestion
    OPTICK_CATEGORY("Draw", Optick::Category::Rendering);
```

File: Source/src/components/ComponentText.cpp [hunk 7]
```diff
@@ -83,7 +83,7 @@ void Hachiko::ComponentText::DrawGui()
 
 void Hachiko::ComponentText::Draw(ComponentTransform2D* transform, Program* program)
 {   
-    OPTICK_CATEGORY("Draw", Optick::Category::Rendering);
+    //OPTICK_CATEGORY("Draw", Optick::Category::Rendering);
```
Comment (baransrc) [inline Source/src/components/ComponentText.cpp:86]:
```suggestion
    OPTICK_CATEGORY("Draw", Optick::Category::Rendering);
```

**Review by monikamnc — APPROVED** [review]
Everything looks OK, can't test game scene because I again have broke metas / lib XD
Thank all the team that made possible VS2
I love you chads <3

**Review by Bernatmago — APPROVED** [review]

**Pull request merged**